tests/postgresql_test.go

```diff
@@ -46,23 +51,76 @@ func TestMain(m *testing.M) {
 	os.Exit(result)
 }
 
-func TestSuccessConnection(t *testing.T) {
-	hostname := "db"
+func TestSuccessConnection90(t *testing.T) {
+	hostname := "db90"
+	envVars := []string{
+		fmt.Sprintf("HOSTNAME=%s", hostname),
+		"USERNAME=postgres",
+		"PASSWORD=example",
+		"DATABASE=demo",
+		"COLLECTION_LIST=ALL",
+		"METRIC=true",
+	}
+	stdout, _, err := executeDockerCompose(containerName, envVars)
+	assert.Nil(t, err)
+	assert.NotEmpty(t, stdout)
+	err = validateJSONSchema(schemav90, stdout)
+	assert.Nil(t, err)
+}
```

Comment:
Instead of duplicating the code for each version, we could use some form of table testing, where you iterate over a predefined array which contains the things that change and the expectations. See, for example, this: https://github.com/golang/go/wiki/TableDrivenTests#parallel-testing

Here they also use `t.Parallel()` which, unsurprisingly, allow tests to run in parallel :P

Reply:
Good Call